src/AssemblerReadGraph.cpp

```diff
@@ -161,116 +161,142 @@ void Assembler::checkReadGraphIsOpen()
 }
 
 
+// Create a local subgraph of the global read graph,
+// starting at a given vertex and extending out to a specified
+// distance (number of edges).
+bool Assembler::createLocalReadGraph(
+        OrientedReadId start,
+        uint32_t maxDistance,           // How far to go from starting oriented read.
+        bool allowChimericReads,
+        bool allowCrossStrandEdges,
+        size_t maxTrim,                 // Used to define containment.
+        double timeout,                 // Or 0 for no timeout.
+        LocalReadGraph& graph)
+{
+    vector<OrientedReadId> starts = {start};
+    bool success = createLocalReadGraph(
+            starts,
+            maxDistance,           // How far to go from starting oriented read.
+            allowChimericReads,
+            allowCrossStrandEdges,
+            maxTrim,                 // Used to define containment.
+            timeout,                 // Or 0 for no timeout.
+            graph
+    );
+
+    return success;
+}
+
 
 // Create a local subgraph of the global read graph,
 // starting at a given vertex and extending out to a specified
 // distance (number of edges).
 bool Assembler::createLocalReadGraph(
-    OrientedReadId start,
+    vector<OrientedReadId>& starts,
     uint32_t maxDistance,           // How far to go from starting oriented read.
     bool allowChimericReads,
     bool allowCrossStrandEdges,
     size_t maxTrim,                 // Used to define containment.
     double timeout,                 // Or 0 for no timeout.
     LocalReadGraph& graph)
 {
-    const auto startTime = steady_clock::now();
+    for (auto& start: starts) {
+        const auto startTime = steady_clock::now();
 
-    // If the starting read is chimeric and we don't allow chimeric reads, do nothing.
-    if(!allowChimericReads && readFlags[start.getReadId()].isChimeric) {
-        return true;
-    }
+        // If the starting read is chimeric and we don't allow chimeric reads, do nothing.
+        if (!allowChimericReads && readFlags[start.getReadId()].isChimeric) {
+            return true;
+        }
 
-    // Add the starting vertex.
-    graph.addVertex(start, uint32_t(markers[start.getValue()].size()),
-        readFlags[start.getReadId()].isChimeric, 0);
+        // Add the starting vertex.
+        graph.addVertex(start, uint32_t(markers[start.getValue()].size()),
+                        readFlags[start.getReadId()].isChimeric, 0);
 
-    // Initialize a BFS starting at the start vertex.
-    std::queue<OrientedReadId> q;
-    q.push(start);
+        // Initialize a BFS starting at the start vertex.
+        std::queue<OrientedReadId> q;
+        q.push(start);
 
 
 
-    // Do the BFS.
-    while(!q.empty()) {
+        // Do the BFS.
```

Comment:
Wouldn't it be better to do a single BFS, initializing the queue with all of the start vertices?